text/0000-extended-standard-library.md

```diff
@@ -0,0 +1,370 @@
+- Feature Name: Extended Standard Library
+- Start Date: 2025-05-08
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC describes an Extended Standard Library (ESL) for Rust. The ESL will consist of a set of crates providing commonly used functionality, subject to policies which promote excellent security, quality, and ease of use.
+
+# Motivation
+[motivation]: #motivation
+
+Rust programs, on average, have many dependencies. This is largely because the Rust standard library is intentionally limited in scope. As a consequence, Rust programs are currently required to pull in many third-party dependencies to get functionality that is considered table-stakes by 2025 standards.
+
+There are three issues this proposal aims to help address, all of which are related to the large number of independent entities providing crates that are widely depended on in Rust software:
+
+### 1. Impractically-sized web of trust
+
+Reliance on a high number of third-party dependencies creates a large web of trust for any non-trivial program. Given the size of this web of trust, it is practically impossible to be confident about the trustworthiness of the full dependency tree.
+
+Being a _trustworthy_ dependency means it:
+
+* Is developed and maintained by qualified, non-malicious authors
+* Has a responsible vulnerability disclosure policy
+* Has sufficient maintainer bandwidth to publish security updates in a reasonable timeframe
+* Has reasonable code review and testing standards
+* Uses secure authentication and release infrastructure
+* Depends only on libraries that are themselves trustworthy
+* Can be expected to maintain these properties well into the future
+
+Establishing confidence in the above characteristics for any individual dependency requires significant effort; establishing confidence in these for all of a project’s dependencies is impractical. As a result, many projects currently rely on fuzzy heuristics to determine the trustworthiness of a dependency, or even choose to ignore the problem altogether and "hope for the best."
+
+The status quo goes against Rust's goal of empowering everyone to build reliable software. Though it was never intended to be that way, as of today the Rust ecosystem nudges developers towards pulling in more dependencies than they can reasonably vet, which in the end hurts the reliability of the software.
+
+### 2. Subpar developer experience
+
+When figuring out which crate to use for relatively basic functionality, there are often many options to choose from with no clear indication of which one should be preferred and why. This makes the whole process difficult, time consuming, and generally unpleasant. The situation detracts significantly from the Rust developer experience.
```

Comment:
I don't understand why choosing a single crate for each purpose is important here. Or maybe I've misunderstood the RFC, and "which one should be preferred" is much more granular, and "not duplicating" is narrowly defined.

For example, could there be multiple async executors (for different use cases), such as performance, embedded, and ergonomics?
Or multiple CLI parsers with different feature sets?